docs/python/tutorials/testing.md

```diff
@@ -0,0 +1,154 @@
+---
+sidebar_position: 20
+title: Testing Your App
+---
+
+### Testing DBOS Functions
+
+Because DBOS workflows, steps, and transactions are ordinary Python functions, you can unit test them using any Python testing framework, like [pytest](https://docs.pytest.org/en/stable/) or [unittest](https://docs.python.org/3/library/unittest.html).
+
+**You must reset the DBOS runtime between each test like this:**
+
+```python
+def reset_dbos():
+    DBOS.destroy()
+    DBOS()
+    DBOS.launch()
+```
+
+This function first cleans up any existing DBOS instance (for example, left over from a previous test or an imported file), then initializes and launches DBOS.
+
+For example, if using pytest, declare `reset_dbos` as a fixture and require it from every test of a DBOS function:
+
+
+```python title="conftest.py"
+import pytest
+from dbos import DBOS
+
+@pytest.fixture()
+def reset_dbos():
+    DBOS.destroy()
+    DBOS()
+    DBOS.launch()
+```
+
+```python title="test_example.py"
+from example_app.main import example_workflow
+
+def test_example_workflow(reset_dbos):
+    example_input = ...
+    example_output = ...
+    assert example_workflow(example_input) == example_output
+
+```
+
+### Custom Configuration
+
+You may want to use a custom configuration of DBOS for testing.
+For example, you likely want to test your application using an isolated development database.
+To do this, save your custom configuration to a file (for example, `dbos-config.testing.yaml`), load it, and pass it to the DBOS object used by your tests:
+
+```python
+def reset_dbos():
+    DBOS.destroy()
+    config = load_config("dbos-config.testing.yaml")
+    DBOS(config=config)
+    DBOS.launch()
+```
+
+Alternatively, you can load your default config then modify its values programatically:
+
+```python
+def reset_dbos():
+    DBOS.destroy()
+    config = load_config()
+    config["database"]["app_db_name"] = f"{config["database"]["app_db_name"]}_test"
+    DBOS(config=config)
+    DBOS.launch()
+```
+
+### Resetting Your Database For Testing
```

Comment:
Next step: add a reset function for system DB. This details block can also be improved after we update the way we manage DB connections.